packages/plugins/documentation/server/services/helpers/utils/clean-schema-attributes.js

```diff
@@ -86,43 +89,49 @@ const cleanSchemaAttributes = (attributes, { typeMap = new Map(), isRequest = fa
       }
       case 'component': {
         const componentAttributes = strapi.components[attribute.component].attributes;
-
+        const rawComponentSchema = {
+          type: 'object',
+          properties: {
+            ...(isRequest ? {} : { id: { type: 'string' } }),
+            ...cleanSchemaAttributes(componentAttributes, {
+              typeMap,
+              isRequest,
+            }),
+          },
+        };
+        const refComponentSchema = {
+          $ref: `#/components/schemas/${`${pascalCase(attribute.component)}Component`}`,
```

Comment:
I think this can be simplified
```suggestion
          $ref: `#/components/schemas/${pascalCase(attribute.component)}Component`,
```